File: src/main/java/ru/citeck/ecos/ui/PasswordCellRenderer.java
package ru.citeck.ecos.ui;

import javax.swing.table.DefaultTableCellRenderer;

public class PasswordCellRenderer
        extends DefaultTableCellRenderer {

    public PasswordCellRenderer() {
        super();
    }

    @Override
    protected void setValue(Object value) {
        int len = value == null? 0: value.toString().length();
        super.setValue("*".repeat(len));
    }

}

File: src/main/kotlin/ru/citeck/idea/view/table/PasswordCellRenderer.kt
package ru.citeck.idea.view.table

import javax.swing.table.DefaultTableCellRenderer

class PasswordCellRenderer : DefaultTableCellRenderer() {

    override fun setValue(value: Any?) {
        val len = value?.toString()?.length ?: 0
        super.setValue("*".repeat(len))
    }
}
